libtiledbsoma/test/unit_soma_geometry_dataframe.cc

```diff
@@ -0,0 +1,149 @@
+                 {.name = "y",
+                  .tiledb_datatype = TILEDB_FLOAT64,
+                  .dim_max = 100,
+                  .string_lo = "N/A",
+                  .string_hi = "N/A",
+                  .use_current_domain = use_current_domain})});
+
+        std::vector<helper::AttrInfo> attr_infos({helper::AttrInfo(
+            {.name = "quality", .tiledb_datatype = TILEDB_FLOAT64})});
+
+        // Check the point cloud doesn't exist yet.
+        REQUIRE(!SOMAGeometryDataFrame::exists(uri, ctx));
+
+        // Create the point cloud.
+        auto [schema, index_columns] =
+            helper::create_arrow_schema_and_index_columns(
+                dim_infos, attr_infos);
+        auto spatial_columns = helper::create_column_index_info(
+            spatial_dim_infos);
+
+        SOMAGeometryDataFrame::create(
+            uri,
+            std::move(schema),
+            ArrowTable(
+                std::move(index_columns.first),
+                std::move(index_columns.second)),
+            ArrowTable(
+                std::move(spatial_columns.first),
+                std::move(spatial_columns.second)),
+            ctx,
+            platform_config,
+            std::nullopt);
+
+        // Check the point cloud exists and it cannot be read as a different
+        // object.
+        REQUIRE(SOMAGeometryDataFrame::exists(uri, ctx));
+        REQUIRE(!SOMASparseNDArray::exists(uri, ctx));
+        REQUIRE(!SOMADenseNDArray::exists(uri, ctx));
+        REQUIRE(!SOMADataFrame::exists(uri, ctx));
+
+        auto soma_geometry = SOMAGeometryDataFrame::open(
+            uri,
+            OpenMode::read,
+            ctx,
+            {},  // column_names,
+            ResultOrder::automatic,
+            std::nullopt);
+        REQUIRE(soma_geometry->uri() == uri);
+        REQUIRE(soma_geometry->ctx() == ctx);
+        REQUIRE(soma_geometry->type() == "SOMAGeometryDataFrame");
+        std::vector<std::string> expected_index_column_names = {
+            dim_infos[0].name,
+            "tiledb__internal__" + spatial_dim_infos[0].name + "__min",
+            "tiledb__internal__" + spatial_dim_infos[1].name + "__min",
+            "tiledb__internal__" + spatial_dim_infos[0].name + "__max",
+            "tiledb__internal__" + spatial_dim_infos[1].name + "__max"};
+
+        std::vector<std::string> expected_spatial_column_names = {
+            spatial_dim_infos[0].name, spatial_dim_infos[1].name};
+        REQUIRE(
+            soma_geometry->index_column_names() == expected_index_column_names);
+        REQUIRE(
+            soma_geometry->spatial_column_names() ==
+            expected_spatial_column_names);
+        REQUIRE(soma_geometry->nnz() == 0);
+        soma_geometry->close();
+
+        auto soma_object = SOMAObject::open(uri, OpenMode::read, ctx);
+        REQUIRE(soma_object->uri() == uri);
+        REQUIRE(soma_object->type() == "SOMAGeometryDataFrame");
+        soma_object->close();
+    }
+}
```

Comment:
seems like we could use some tests which write & read data -- is this planned for a future PR?

Reply:
Yes, there is #3222 which adds tests for read/write.